flink-libraries/flink-table/src/test/scala/org/apache/flink/api/scala/batch/sql/SelectITCase.scala

```diff
@@ -146,4 +146,22 @@ class SelectITCase(
     tEnv.sql(sqlQuery)
   }
 
+  @Test
+  def testSelectWithCompositeType(): Unit = {
+    val env = ExecutionEnvironment.getExecutionEnvironment
+    val tEnv = TableEnvironment.getTableEnvironment(env, config)
+
+    val sqlQuery = "SELECT MyTable.a2, MyTable.a1._2 FROM MyTable"
```

Comment:
I found leaving out the table name (or table alias) leads to a Calcite exception which is hard to understand. Can we throw a `TableException` instead which is more helpful for the user?